File: src/Jenin.FontAwesome.Blazor/wwwroot/js/fontawesome-config-utilities.js
"use strict";

class FontAwesomeConfigUtilities {
    setAutoReplaceSvg(value) {
        if (window.FontAwesomeConfig) {
            window.FontAwesomeConfig.autoReplaceSvg = value;
        }
    };

    getAutoReplaceSvg() {
        if (window.FontAwesomeConfig) {
            return window.FontAwesomeConfig.autoReplaceSvg;
        }

        return null;
    };
}

window["FontAwesomeConfigUtilities"] = new FontAwesomeConfigUtilities();

File: src/Jenin.FontAwesome.Blazor/JsInterop/FontAwesomeConfigUtilities.cs
namespace Jenin.FontAwesome.Blazor.JsInterop;

internal static class FontAwesomeConfigUtilities {
    public static async ValueTask<string> SetAutoReplaceSvgAsync(this IJSRuntime jSRuntime, string value) {
        var old = await jSRuntime.InvokeAsync<string>("FontAwesomeConfigUtilities.getAutoReplaceSvg");
        await jSRuntime.InvokeVoidAsync("FontAwesomeConfigUtilities.setAutoReplaceSvg", value);

        return old;
    }

    public static ValueTask<string> GetAutoReplaceSvgAsync(this IJSRuntime jSRuntime)
        => jSRuntime.InvokeAsync<string>("FontAwesomeConfigUtilities.getAutoReplaceSvg");
}
